csharp/src/Microsoft.ML.OnnxTraining/TrainingParameters.cs

```diff
@@ -0,0 +1,336 @@
+﻿using Microsoft.ML.OnnxRuntime;
+using Microsoft.ML.OnnxRuntime.Tensors;
+using System;
+using System.Buffers;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+using System.Text;
+
+namespace Microsoft.ML.OnnxTraining
+{
+    /// <summary>
+    /// Holds the training parameters used by the training runner.
+    /// </summary>
+    public class TrainingParameters : SafeHandle
```

Comment:
>SafeHandle [](start = 38, length = 10)

It has disposable members but no native handles. Please, implement IDisposable

Reply:
I have changed this to mirror your new implementation of InferenceSession.cs by removing the SafeHandle override and adding a new _nativeHandle.  When you are ready, the next update will have this.